test/privateIP.js

```diff
@@ -0,0 +1,213 @@
+var sinon = require('sinon'),
+    assert = require('assert'),
+    request = require('request-promise'),
+    OpenAMClient = require('../lib/openam').OpenAMClient;
+
+describe('Test with private IP', function () {
+    var hostname, serverUrl, privateIP;
+
+    beforeEach(function () {
+
+        serverUrl = 'http://openam.example.com:8080/openam';
+        hostname = 'openam.example.com';
+        privateIP = '127.0.0.1';
+
+        // client instance
+        clientIP = new OpenAMClient(serverUrl, privateIP);
+
+        // spies
+        getRequests = sinon.spy(request, 'get');
+        postRequests = sinon.spy(request, 'post');
+
```

Comment:
replace these with

```javascript
getRequests = sinon.stub(request, 'get');
postRequests = sinon.stub(request, 'post');
```

-- that fixed the unit tests for me.

(I think `spy` still invokes the original method whereas `stub` doesn't)